vector_double.cpp

```diff
@@ -0,0 +1,192 @@
+#include "vector_double.hpp"
+
+// Copy Constructor
+template<class T>
+Vector<T>::Vector(const Vector<T>& b)
+{
+    Vcapacity = b.space;
+    size_v = b.size_v;
+    elem = b.elem;
+
+}
+template<class T>
+Vector<T>::Vector(Vector<T>&& obj)
+{
+    Vcapacity = obj.space;
+    size_v = obj.size_v;
+    elem = obj.elem;
+    obj.elem = nullptr;
+    obj.elem = nullptr;
+}
+
+//explicit
+template<class T>
+Vector<T>::Vector(unsigned int)
+{
+    size_v = 0;
+    Vcapacity  = 0;
+    elem   = nullptr;
+}
+
+
+template<class T>
+Vector<T>::~Vector()
+{
+    delete[] elem;
+}
+
+template<class T>
+int Vector<T>::size() const
+{
+    return size_v;
+}
+
+template<class T>
+int Vector<T>:: capacity()const
+{
+    return Vcapacity;
+}
+
+template<typename T>
+void Vector<T>::push_back(const T &v)
+{
+
+    if (size_v >= Vcapacity)
+    {
+        reserve(Vcapacity*2);
+        elem [size_v] = v;
```

Comment:
this needs to happen outside of the if statement, and size needs to get incremented